src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<? extends Machine> machines = new BulldozerProducer().get();
```

Comment:
Create additional class variables for BulldozerProducer, ExcavatorProducer, and TruckProducer, then return simple bulldozerProducer.get(), ... instead of creating a new instance every time

Reply:
I can't do this way, because my buldozerProducer.get() method will return List of Buldozers, truckProducer.get() will return List of Trucks etc. But this method should return List of Machines, it will not compile. That's why I create List with wildcard that extends any Machine class and return it as List of Machines, as it requires.